### ISSUE TITLE
v1.9.0 doesn't build std_names (encoding exception)

### ISSUE THREAD

**Issue by pelson** [SciTools/iris#1876]
```
bin/python tools/generate_std_names.py etc/cf-standard-name-table.xml build/lib/iris/std_names.py
Traceback (most recent call last):
  File "tools/generate_std_names.py", line 128, in <module>
    to_dict(args.input, args.output)
  File "tools/generate_std_names.py", line 102, in to_dict
    tree = ET.parse(infile)
  File "lib/python3.5/xml/etree/ElementTree.py", line 1182, in parse
    tree.parse(source, parser)
  File "lib/python3.5/xml/etree/ElementTree.py", line 594, in parse
    self._root = parser._parse_whole(source)
  File "lib/python3.5/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 1162: ordinal not in range(128)
error: command 'bin/python' failed with exit status 1

```

Fix incoming.

**Comment by pelson** [comment]
Looks like this is a locale problem. On the machine in question:

```
python -c "import locale; print(locale.getpreferredencoding())"
ANSI_X3.4-1968
```

Fixed this by setting:

```
export LC_ALL="en_GB.UTF-8"
```

Perhaps we should be asserting that the file being read is UTF-8 (which it is, as we are talking about the standard name table).

**Comment by pelson** [comment]
@QuLogic - do you have any thoughts on this?

**Comment by QuLogic** [comment]
If you're reading XML with `lxml`, they recommend using a binary file (`rb`) so that it can read the encoding from the file and decode it correctly.

**Comment by QuLogic** [comment]
Opening the output file as binary and encoding before writing is probably also the easiest fix there, too.

**Comment by QuLogic** [comment]
Fixed by #1878?

**Comment by pelson** [comment]
> Fixed by #1878?

:+1:
